web/src/engine/keyboard/src/keyboards/keyboardLoaderBase.ts

```diff
@@ -33,17 +41,22 @@ export abstract class KeyboardLoaderBase {
    * @param stub  The stub of the keyboard to load.
    * @returns     A Promise that resolves to the loaded keyboard.
    */
-  public async loadKeyboardFromStub(stub: KeyboardStub): Promise<JSKeyboard> {
+  public async loadKeyboardFromStub(stub: KeyboardStub): Promise<Keyboard> {
     this.harness.install();
     return this.loadKeyboardInternal(stub.filename, new StubBasedErrorBuilder(stub));
   }
 
-  private async loadKeyboardInternal(uri: string, errorBuilder: KeyboardLoadErrorBuilder): Promise<JSKeyboard> {
+  private async loadKeyboardInternal(uri: string, errorBuilder: KeyboardLoadErrorBuilder): Promise<Keyboard> {
     const byteArray = await this.loadKeyboardBlob(uri, errorBuilder);
 
     if (byteArray.slice(0, 4) == Uint8Array.from([0x4b, 0x58, 0x54, 0x53])) { // 'KXTS'
       // KMX or LDML (KMX+) keyboard
-      console.error("KMX keyboard loading is not yet implemented!");
+      const result = (await this._km_core).keyboard_load_from_blob(uri, byteArray);
+      if (result.status == KM_CORE_STATUS.OK) {
+        // extract keyboard name from URI
+        const id = uri.split('#')[0].split('?')[0].split('/').pop().split('.')[0];
```

Comment:
Is the keyboard name guaranteed to be in the URI?

Reply:
Good point. I'll address this in a follow-up PR.